src/common/iceberg.cpp

```diff
@@ -8,45 +8,121 @@
 
 namespace duckdb {
 
-template <class OP>
-static void ReadManifestEntries(ClientContext &context, const vector<IcebergManifest> &manifests,
-                                bool allow_moved_paths, FileSystem &fs, const string &iceberg_path,
-                                vector<IcebergTableEntry> &result) {
-	for (auto &manifest : manifests) {
-		auto manifest_entry_full_path = allow_moved_paths
-		                                    ? IcebergUtils::GetFullPath(iceberg_path, manifest.manifest_path, fs)
-		                                    : manifest.manifest_path;
-		auto manifest_paths = ScanAvroMetadata<OP>("IcebergManifest", context, manifest_entry_full_path);
-		result.push_back({std::move(manifest), std::move(manifest_paths)});
-	}
-}
-
 IcebergTable IcebergTable::Load(const string &iceberg_path, IcebergSnapshot &snapshot, ClientContext &context,
                                 const IcebergOptions &options) {
 	IcebergTable ret;
 	ret.path = iceberg_path;
 	ret.snapshot = snapshot;
 
-	auto &fs = FileSystem::GetFileSystem(context);
-	auto manifest_list_full_path = options.allow_moved_paths
-	                                   ? IcebergUtils::GetFullPath(iceberg_path, snapshot.manifest_list, fs)
-	                                   : snapshot.manifest_list;
 	vector<IcebergManifest> manifests;
+	unique_ptr<ManifestReader> manifest_reader;
```

Comment:
These changes are just to get this code more in line with the version of this for the `iceberg_scan` path